velox/docs/functions/spark/string.rst

```diff
@@ -53,6 +53,13 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
         SELECT conv("11abc", 10, 10); -- '11'
         SELECT conv('H016F', 16, 10); -- '0'
 
+.. spark:function:: empty2null(input) -> varchar
+
+    Returns NULL if ``input`` is empty. Otherwise, returns ``input``. ::
```

Comment:
Perhaps, mention that this is an internal Spark function used in ...

Reply:
how about
```
Returns NULL if ``input`` is empty. Otherwise, returns ``input``.
Note: it's an internal Spark function used in V1Writes. ::
```

Reply:
Sure. What is V1Writes ? Can we add a link to allow reader to get more context easily?

Reply:
> What is V1Writes ?

hmm.. it's a version of Spark Datasource interface.

> Can we add a link to allow reader to get more context easily?

can I add this link in docs? https://github.com/apache/spark/blob/34a52ad2ce377dd077c4cf70740ba3ab6c18c739/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L3486-L3508 , there is no more explanation for this function, but link of source code may invalid if code refactoring happens..

I suggest just add notes of it's an internal Spark function without detailed usage, how do you feel about this?

Reply:
I see. Would it be possible to provide some high-level explanation of where it is used? 

> Spark V1Writers add Empty2Null to project operator, 

Is empty2null used to convert empty strings intro nulls when writing strings to Parquet / ORC files? What's the point of this conversion?

Reply:
Spark use `Empty2Null` to convert empty value to NULL for string type partition column, then it can be converted to hive default partition value `__HIVE_DEFAULT_PARTITION__`

Reply:
I see. Thank you for clarifying. Sounds similar to https://www.ibm.com/docs/en/db2-big-sql/6.0?topic=dtidbs-how-hive-handles-null-values-partitioning-column-type-varchar

It would be nice to include this explanation in the docs to provide some more context to a reader.

```
Spark uses empty2null to convert empty value of a partition column, which is then converted to Hive default partition value __HIVE_DEFAULT_PARTITION__.
```

Interesting that Spark chooses to convert values in 2 steps: (1) empty -> NULL; (2) NULL -> __HIVE_DEFAULT_PARTITION__.

Reply:
done.